apps/zui/scripts/merge-mac-release-files.mjs

```diff
@@ -0,0 +1,227 @@
+// Based on https://github.com/electron-userland/electron-builder/issues/5592#issuecomment-2004803764
+import { Octokit } from "octokit";
+import pkg from '../package.json' assert {type: 'json'};
+import { TextDecoder } from 'node:util';
+import fs from 'node:fs';
+import { Readable } from 'node:stream';
+import { Buffer } from 'node:buffer';
+import yaml from 'js-yaml';
+
+const token = process.env.GH_TOKEN
+
+const client = new Octokit({
+  auth: token
+});
+
+// These are derived from settings in package.json so the script will work on
+// both regular Zui and Zui Insiders.
+const OWNER = pkg.repository.split('/')[3];
+const REPO = pkg.repository.split('/')[4];
+const PRODUCT_NAME= pkg.productName.replaceAll(' ', '-');
+const REPO = 'zui';
+const URL = `/repos/${OWNER}/${REPO}/releases`;
+const VERSION = pkg.version;
+const RELEASE_NAME = (PRODUCT_NAME == 'Zui') ? 'v' + VERSION : VERSION;
+const FILE_NAME = 'latest-mac.yml';
+const LOCAL_FILE_PATH = `dist/apps/zui/${FILE_NAME}`;
+
+const mergeFiles = (intel, arm) => {
+  const intelObject = yaml.load(intel);
+  const armObject = yaml.load(arm);
+
+  const mergedObject = {
+    ...intelObject,
+  };
+
+  mergedObject.files = [
+    ...intelObject.files,
+    ...armObject.files,
+  ];
+
+  const dumpOptions = {
+    // avoids moving the sha512 checksum into its own line
+    lineWidth: -1,
+  };
+
+  const mergedString = yaml.dump(mergedObject, dumpOptions);
+  return mergedString;
+}
+
+const getPlatformFromLatestMacYml = (content) => {
+  const intelRe = `${PRODUCT_NAME}-${VERSION}.dmg`
```

Comment:
Is there anyway that we could have the intel version be labled `*-amd64.dmg` or `*-intel64.dmg` (vlc calls its intel64/arm64)?

Reply:
@mattnibs: I looked into it and did find a way, so I've just pushed a commit that makes the change. I just tested it out in my personal fork and you can see the improvement in https://github.com/philrz/zui/releases/tag/v1.7.5. I just tested it out in my Intel Mac by starting from the previous release https://github.com/philrz/zui/releases/tag/v1.7.4 when the filename was just `Zui-1.7.4.dmg` and it auto-updated to the newer release with filename `Zui-1.7.5-x64.dmg` and restarted without complaint. If you'd like to repeat that test with the arm64 builds on your Apple Silicon Mac that's be great, but I'm planning on a final test after I've resolved all PR feedback, so, whenever. I see no reason why it wouldn't work since this change should have been a no-op for those builds, but I've seen some wacky stuff on this journey, so I've become quite paranoid. 😛